### PR TITLE
[Improve][Connector-V2][Iotdb] Unified exception for iotdb source & sink connector

### LINKED ISSUE #3536: [Improve][Connector-V2][Iotdb] Unified exception for iotdb source & sink connector
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

[Improve][Connector-V2][Iotdb] Unified exception for iotdb source & sink connector

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wuchunfu**
## Purpose of this pull request

sub-task #2930 

close #3536 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by wuchunfu** [comment]
@TyrantLucifer @Hisoka-X PTAL, thanks

**Comment by wuchunfu** [comment]
@EricJoy2048 PTAL,thanks

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/sink/IoTDBSinkClient.java [hunk 1]
```diff
@@ -144,30 +148,31 @@ synchronized void flush() throws IOException {
...
                 try {
                     long backoff = Math.min(sinkConfig.getRetryBackoffMultiplierMs() * i,
-                            sinkConfig.getMaxRetryBackoffMs());
+                        sinkConfig.getMaxRetryBackoffMs());
                     Thread.sleep(backoff);
                 } catch (InterruptedException ex) {
                     Thread.currentThread().interrupt();
-                    throw new IOException(
-                            "Unable to flush; interrupted while doing another attempt.", e);
+                    throw new IotdbConnectorException(IotdbConnectorErrorCode.FLUSH_RECORDS_FAILED,
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/sink/IoTDBSinkClient.java]: CommonErrorCode.FLUSH_DATA_FAILED is better

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/sink/IoTDBSinkClient.java [hunk 2]
```diff
@@ -177,17 +182,18 @@ synchronized void flush() throws IOException {
 
     private void checkFlushException() {
         if (flushException != null) {
-            throw new RuntimeException("Writing records to IoTDB failed.", flushException);
+            throw new IotdbConnectorException(IotdbConnectorErrorCode.WRITING_RECORDS_FAILED,
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/sink/IoTDBSinkClient.java]: CommonErrorCode.FLUSH_DATA_FAILED is better

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/sink/IoTDBSinkClient.java (hunk 1 above)
Comment (wuchunfu) [reply seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/sink/IoTDBSinkClient.java]: Done, Thank you for your reminder

**Comment by wuchunfu** [comment]
@TyrantLucifer @CalvinKirs @Hisoka-X PTAL, thanks

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java [hunk 3]
```diff
@@ -64,7 +66,11 @@ public void prepare(Config pluginConfig) throws PrepareFailException {
             result = CheckConfigUtil.checkAllExists(pluginConfig, NODE_URLS.key());
 
             if (!result.isSuccess()) {
-                throw new PrepareFailException(getPluginName(), PluginType.SOURCE, "host and port and node urls are both empty");
+                throw new IotdbConnectorException(SeaTunnelAPIErrorCode.CONFIG_VALIDATION_FAILED,
+                    String.format("PluginName: %s, PluginType: %s, Message: %s",
+                        getPluginName(), PluginType.SOURCE,
+                        "host and port and node urls are both empty")
```
Comment (TaoZex) [inline seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java]: How about define string as constants separately?

**Review by TyrantLucifer — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java (hunk 3 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java]: I think define string as constants is meaningless, static variables are defined for reuse.

**Review by Hisoka-X — DISMISSED** [review]

**Review by Hisoka-X — APPROVED** [review]

File: seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java (hunk 3 above)
Comment (wuchunfu) [reply seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSource.java]: Done, Thank you for your reminder

**Review by TaoZex — APPROVED** [review]

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**